property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
+        break;
+      case 'date':
+        console.warn('Warning: property default value was converted' +
+          ' from string using ISO formatting');
+        if (value.toLowerCase() === 'now'){
+          def.defaultFn = 'now';
+        } else {
+          def.default = new Date(value);
+        }
+        break;
+      case 'geopoint':
+        if (value.indexOf('lat') !== -1 && value.indexOf('lng') !== -1) {
+          def.default = JSON.parse(value);
+        } else {
+          var geo = value.replace(/[\s,]+/g, ',').split(',');
+          def.default = {};
+          def.default.lat = Number(geo[0]);
+          def.default.lng = Number(geo[1]);
+        }
+        break;
+      case 'buffer':
+        console.warn('Warning: property default value was converted' + 
+          'from string using UTF8 encoding');
```

Comment:
@bajtos, @alexpit,

I suggested UTF8 because it is the default for the `Buffer` NodeJS class. I think it's fair to assume that a LoopBack Buffer would use the same defaults as NodeJS.

If the user would like to use a different encoding, they can input the default value directly into the JSON (which is what I would assume most users would do in the first place even if they wanted UTF8).